### PR TITLE
Ensure connectivity check listens for gain of connection

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Sets the default value for `bsg_current_reachability_state` to -1. This prevents the connectivity listener from not firing when a connection is regained, as 0 is a valid value for the bitmask cached later in the method.

### CODE REVIEW

File: Bugsnag/Delivery/BSGConnectivity.m [hunk 1]
```diff
@@ -51,11 +51,13 @@ BOOL BSGConnectivityShouldReportChange(SCNetworkReachabilityFlags flags) {
...
     // Check if the reported state is different from the last known state (if any)
-    if ((flags & importantFlags) != (bsg_current_reachability_state & importantFlags)) {
+    SCNetworkReachabilityFlags newFlags = flags & importantFlags;
+    SCNetworkReachabilityFlags oldFlags = bsg_current_reachability_state & importantFlags;
+    if (newFlags != oldFlags) {
         // When first subscribing to be notified of changes, the callback is
         // invoked immmediately even if nothing has changed. So this block
         // ignores the very first check, reporting all others.
-        if (bsg_current_reachability_state == 0) {
+        if (bsg_current_reachability_state == -1) {
```
Comment (tomlongridge) [inline Bugsnag/Delivery/BSGConnectivity.m:60]: For this to work, don't we need to initialize `bsg_current_reachability_state` to -1 to catch the first call after registering?

Comment (fractalwrench) [reply Bugsnag/Delivery/BSGConnectivity.m:60]: Yes - this was a local change that didn't make its way into this changeset, I've updated the PR.

**Comment by fractalwrench** [comment]
Additionally removed one unit test that is inherently flaky on CI due to the use of global mutable state in the implementation.

**Comment by Cawllec** [comment]
Manually verified on a device that both the Connectivity Breadcrumb is appearing when WI-FI is enabled, and that delivery occurs when the connectivity is re-enabled.

**Pull request merged**